### PR TITLE
DOCS: General intro to Mumble's source code

### PR DESCRIPTION
**Pull request by Krzmbrzl**
### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

File: docs/dev/TheMumbleSourceCode.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+the implementation of that function is also in the source file with the corresponding name. So it can happen that the file `MyClass.h` defines a class
+`MyClass` that declares a function `X`. That function may not end up being implemented in `MyClass.cpp`. Classes that make extensive use of this
+pattern are the `Server` and `MainWindow` class respectively. If the header doesn't mention where the given set of functions is implemented, your best
+chance is to search the source files in the same directory for a matching implementation (in almost all cases, the implementation lives in a source
+file within the same directory as the header in question).
+
+
+### Client
+
+First off: all the `*.ts` are used for localizations (translations) are are handled by an external services. Thus, you should not modify them by hand
```
Comment (klemensn) [inline docs/dev/TheMumbleSourceCode.md:111]: s/an//

File: docs/dev/TheMumbleSourceCode.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+`samples` and `screenshots` are self-describing and generally of very little interest when dealing with the Mumble source code as such. `scripts` is
+mainly occupied by various scripts used throughout various tasks surrounding Mumble. In addition to that, it also contains a few things such as a
+default configuration file for the Mumble server.
+
+Your main focus should lie on the `src` directory as this is where the bulk of Mumble's source code is living. Directly in `src/` are shared sources
+that are used by the Mumble client as well as by the server that live in `src/mumble` and `src/murmur` respectively. The `src/mumble_exe` direcrtory
+contains code for the Windows-specific executable. In order to make sense of that you have to know that on Windows we are currently using a bit of an
+odd system: The main application is packaged inside a DLL that is loaded by the main executable. This is a historic artifact and currently only
+servers to create problems every now and then. In any case though, the mentioned "main executable" whose sole job is loading of that DLL lives in
```
Comment (klemensn) [inline docs/dev/TheMumbleSourceCode.md:88]: s/servers/serves/

**Review by klemensn — COMMENTED** [review]
Much appreciated.

**Comment by NormandoHall** [comment]
This is a GREAT collaboration to incentive developers to involved more. Thanks!

**Comment by Krzmbrzl** [comment]
Good to see that it is helpful :)

**Pull request merged**